src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/DatabaseService.java

```diff
@@ -35,18 +38,62 @@
  */
 public interface DatabaseService<MODEL, HASH, ACCESSION> {
 
-    List<AccessionWrapper<MODEL, HASH, ACCESSION>> findAllAccessionsByHash(Collection<HASH> hashes);
+    /**
+     * Finds all valid accessioned model data that has a hashed message in the collection @param hashes.
+     *
+     * @param hashes
+     * @return
+     */
+    List<ModelWrapper<MODEL, HASH, ACCESSION>> findAllModelByHash(Collection<HASH> hashes);
+
+    /**
+     * @param accession
+     * @return Active accession with versioning data.
+     * @throws AccessionDoesNotExistException when accession does not exist.
+     * @throws AccessionMergedException       when accession has been merged with another one, its accession id is included
+     *                                        in the exception.
+     * @throws AccessionDeprecatedException   accession is no longer active.
+     */
+    AccessionWrapper<MODEL, HASH, ACCESSION> findAccession(ACCESSION accession) throws
+            AccessionDoesNotExistException, AccessionMergedException, AccessionDeprecatedException;
+
+    /**
+     * Finds last version of all valid accessions with their possible data model representations.
+     *
+     * @param accessions valid accession id
+     * @return All valid accessions. No deprecated or merged ids will be returned.
+     */
+    List<ModelWrapper<MODEL, HASH, ACCESSION>> findAllAccessions(List<ACCESSION> accessions);
+
+    /**
+     * Finds a specific version of accession and their data model representations.
+     *
+     * @param accession
+     * @param version
+     * @return
+     * @throws AccessionDoesNotExistException when accession does not exist.
+     * @throws AccessionMergedException       when accession has been merged with another one, its accession id is included
+     *                                        in the exception.
+     * @throws AccessionDeprecatedException   accession is no longer active.
+     */
+    ModelWrapper<MODEL, HASH, ACCESSION> findAccessionVersion(ACCESSION accession, int version)
```

Comment:
This method name is a bit confusing to me. It sounds like trying to get the version of an accession, which is not the case. I think it is better to call findAccession(ACCESSION, int) which overloads findAccession(ACCESSION) or findByAccessionAndVersion(ACCESSION, int)

Reply:
I'll use the @cyenyxe card to avoid changing things until we have consensum.
My two cents, I'm a horrible person for naming and I see your point.